covid-vaccine-faq-bot/functions/es-dialogflow-detect-intent.js

```diff
@@ -0,0 +1,42 @@
+// Imports the Google Cloud API library
+const {SessionsClient} = require('@google-cloud/dialogflow');
+ 
+exports.handler = async function (context, event, callback) {
+ 
+ let query = event.utterance;
+ 
+const client = new SessionsClient({
+  keyFilename: Runtime.getAssets()[context.GOOGLE_APPLICATION_CREDENTIALS].path
+});
+ 
+ // dialogflow keeps conversation state organized by session ids.  in order to have back and forth passes with dialogflow, we need to maintain a consistent
+ // session id throughout the dialog.  if one doesn't exist, generate it.  if it does, use what we have.
+ if (!event.dialogflow_session_id){
+   event.dialogflow_session_id = Math.random().toString(36).substring(7);
+ }
+ 
+ const request = {
+   session: client.projectAgentSessionPath(
+     context.DIALOGFLOW_ES_PROJECT_ID,
+     event.dialogflow_session_id
+   ),
+   queryInput: {
+     text: {
+       text: query,
+       languageCode: 'en-US',
```

Comment:
Should we make `languageCode` an environment variable (and default it to english) so folks can use this for multi-language chatbots?

Reply:
i have made changes please check